### ISSUE TITLE
Invalid redirection target on fish shell

### ISSUE THREAD

**Issue by jkerdreux-imt** [ConnerWill/yayfzf#4]
My default shell is Fish (https://fishshell.com/). Everything is fine except the preview window that show a invalid redirection error.

![image](https://github.com/ConnerWill/yayfzf/assets/4437893/8ab43228-02f5-47fd-817f-919b37d7fd06)

**Comment by ConnerWill** [comment]
@jkerdreux-imt 

# Troubleshooting

## Version

What is the output of this command?
```bash
fish --version && yayfzf --version
```

## Installation Method

How did you install ***yayfzf*** ?

* AUR
* Run directly (`./yayfzf`)
* Sourced file (`source ./yayfzf && yayfzf`)
* Other (Please explain)

## Testing other shells

Have you tested running `yayfzf` with a different shell other than fish? *(bash, zsh, etc.)*

Do you get the same error while using a different shell?

```bash
exec bash
yayfzf
```

```bash
exec zsh
yayfzf
```

## My own testing

I tested `yayfzf` by installing the latest version of ***fish*** with pacman

I am not experiencing the same issue as you are.

```bash
exec fish
yayfzf
```

```bash
fish -c "yayfzf"
```

## Bash installed?

* Do you have bash installed on your system?
* Is bash linked to fish?

Make sure bash is installed on your system and it can be run

```bash
exec bash
```

**Comment by jkerdreux-imt** [comment]
Here my answers : 

**OS:**
Manjaro (Arch Linux based distro)

**Version:** 
fish, version 3.6.1
yayfzf: 1.0.2

**Installation Method** 
- AUR 

**Other shells**
Even if I run yayfzf from bash the error still occurs (same w/ zsh). The main reason : fish is still my default shell ($SHELL=/usr/bin/fish). So when yayfzf run an execl fish is called ..

![Screenshot_2023-07-13_11-22-19](https://github.com/ConnerWill/yayfzf/assets/4437893/21c275d6-1cd2-4a7d-85f9-d77fe06a497d)

Setting $SHELL to bash fix the issue (not really an fantastic way, but it's working).

![Screenshot_2023-07-13_11-27-16](https://github.com/ConnerWill/yayfzf/assets/4437893/a0de6efd-2275-4f16-a0d2-6dad855b8a05)

Hope this helps

**Comment by ConnerWill** [comment]
@jkerdreux-imt 

**Thank you for your detailed response and for your screenshots.**

## Further Testing

I set my default shell to *fish* then ran `yayfzf`, I got the same error you did.

![image](https://github.com/ConnerWill/yayfzf/assets/85781761/3d5a06d4-5cb1-4ce2-bdbf-163aa31e5a09)

The issue seems to be with [Line 288](https://github.com/ConnerWill/yayfzf/blob/main/bin/yayfzf#L288) and with how fish shell handles redirection.


## Searching For A Solution

I am currently working on a permanent solution for this issue. 

This issue will hopefully be resolved in *yayfzf **v1.0.3***

## Temporary Workaround

As a temporary workaround to get you by until this issue has been fixed and v1.0.3 is released...

You can create an alias or function in your shell to run ***yayfzf*** using bash

Here is an example of how to set the alias:

```bash
alias yayfzf="SHELL=$(command -v bash) command yayfzf"
```

Add that alias command to your shell's rc file *(e.g. `~/.config/fish/config.fish`)*

## Contribution

If you are able to troubleshoot and resolve this issue so `yayfzf` can be run using fish shell, I would greatly appreciate it!

Fork this repo, create a new branch for this issue, apply fix to code, then create a pull request. I will merge request if fix resolves this issue.

I myself will continue to try and resolve this issue :) 

---

**Leaving this issue open until issue has been fixed**
